test/src/TestHdlcpp.cpp

```diff
@@ -42,104 +41,104 @@ class HdlcppFixture {
     std::shared_ptr<Hdlcpp::Hdlcpp> hdlcpp;
     std::vector<uint8_t> readBuffer;
     std::vector<uint8_t> writeBuffer;
-    uint8_t buffer[10];
+    uint8_t dataBuffer[10];
 };
 
 TEST_CASE_METHOD(HdlcppFixture, "hdlcpp test", "[single-file]")
 {
     SECTION("Test write with invalid input")
     {
-        CHECK(hdlcpp->write(buffer, 0) == -EINVAL);
-        CHECK(hdlcpp->write(nullptr, sizeof(buffer)) == -EINVAL);
+        CHECK(hdlcpp->write({dataBuffer, 0}) == -EINVAL);
+        CHECK(hdlcpp->write({}) == -EINVAL);
```

Comment:
The default constructor of `std::span` constructs an empty span whose `data() == nullptr` and `size() == 0`. 
https://en.cppreference.com/w/cpp/container/span/span